packages/create-ponder/src/helpers/getEtherscanChainId.ts

```diff
@@ -1,21 +1,67 @@
 const networkByEtherscanHostname: Record<
   string,
-  { name: string; chainId: number } | undefined
+  { name: string; chainId: number; apiUrl: string } | undefined
 > = {
-  "etherscan.io": { name: "mainnet", chainId: 1 },
-  "ropsten.etherscan.io": { name: "ropsten", chainId: 3 },
-  "rinkeby.etherscan.io": { name: "rinkeby", chainId: 4 },
-  "goerli.etherscan.io": { name: "goerli", chainId: 5 },
-  "kovan.etherscan.io": { name: "kovan", chainId: 42 },
-  "sepolia.etherscan.io": { name: "sepolia", chainId: 11155111 },
-  "optimistic.etherscan.io": { name: "optimism", chainId: 10 },
-  "kovan-optimistic.etherscan.io": { name: "optimism-kovan", chainId: 69 },
-  "goerli-optimism.etherscan.io": { name: "optimism-goerli", chainId: 420 },
-  "polygonscan.com": { name: "polygon", chainId: 137 },
-  "mumbai.polygonscan.com": { name: "polygon-mumbai", chainId: 80001 },
-  "arbiscan.io": { name: "arbitrum", chainId: 42161 },
-  "testnet.arbiscan.io": { name: "arbitrum-rinkeby", chainId: 421611 },
-  "goerli.arbiscan.io": { name: "arbitrum-goerli", chainId: 421613 },
+  "etherscan.io": {
+    name: "mainnet",
+    chainId: 1,
+    apiUrl: "https://api.etherscan.io/api",
+  },
+  "ropsten.etherscan.io": {
+    name: "ropsten",
+    chainId: 3,
+    apiUrl: "https://api-ropsten.etherscan.io//api",
+  },
+  "rinkeby.etherscan.io": {
+    name: "rinkeby",
+    chainId: 4,
+    apiUrl: "https://api-rinkeby.etherscan.io/api",
+  },
+  "goerli.etherscan.io": {
+    name: "goerli",
+    chainId: 5,
+    apiUrl: "https://api-goerli.etherscan.io/api",
+  },
+  "kovan.etherscan.io": {
+    name: "kovan",
+    chainId: 42,
+    apiUrl: "https://api-kovan.etherscan.io/api",
+  },
+  "sepolia.etherscan.io": {
+    name: "sepolia",
+    chainId: 11155111,
+    apiUrl: "https://api-sepolia.etherscan.io/api",
+  },
+  "optimistic.etherscan.io": {
+    name: "optimism",
+    chainId: 10,
+    apiUrl: "https://api-optimistic.etherscan.io/api",
+  },
+  "goerli-optimism.etherscan.io": {
+    name: "optimism-goerli",
+    chainId: 420,
+    apiUrl: "https://api-goerli-optimistic.etherscan.io/api",
+  },
+  "polygonscan.com": {
+    name: "polygon",
+    chainId: 137,
+    apiUrl: "https://api.polygonscan.com/api",
+  },
+  "mumbai.polygonscan.com": {
+    name: "polygon-mumbai",
+    chainId: 80001,
+    apiUrl: "https://api-testnet.polygonscan.com/api",
+  },
+  "arbiscan.io": {
+    name: "arbitrum",
+    chainId: 42161,
+    apiUrl: "https://api.arbiscan.io/api",
+  },
+  "goerli.arbiscan.io": {
+    name: "arbitrum-goerli",
+    chainId: 421613,
+    apiUrl: "https://api-goerli.arbiscan.io/api",
+  },
```

Comment:
wonder if wagmi folks would be open to extending their chain database with blockexplorer API URLs:
https://github.com/wagmi-dev/references/blob/3c6104c0c0e96294bd7ab0c35449a437b6e2d38f/packages/chains/src/mainnet.ts